fineract-loan/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -2698,6 +2698,10 @@ public ChangedTransactionDetail adjustExistingTransaction(final LoanTransaction
                     transactionForAdjustment, scheduleGeneratorDTO);
         }
 
+        if (newTransactionDetail.isCreditBalanceRefund()) {
```

Comment:
I dont really like this. It would be better to not throw an ACTIVE status change event and after az OVERPAID status change event.